pkg/module/go_get_fetcher_test.go

```diff
@@ -0,0 +1,43 @@
+package module
+
+import (
+	"github.com/spf13/afero"
+	"io/ioutil"
+)
+
+func (s *ModuleSuite) TestNewGoGetFetcher() {
+	r := s.Require()
+	fs := s.fs
+	fetcher, err := NewGoGetFetcher(s.goBinaryName, fs, repoURI, version)
+	r.NoError(err)
+	goGetFetcher, ok := fetcher.(*goGetFetcher)
+	r.True(ok)
+	r.Equal(repoURI, goGetFetcher.repoURI)
+	r.Equal(version, goGetFetcher.version)
+}
+
+func (s *ModuleSuite) TestGoGetFetcherFetch() {
+	r := s.Require()
+	// we need to use an OS filesystem because fetch executes vgo on the command line, which
+	// always writes to the filesystem
+	fs := afero.NewOsFs()
+	fetcher, err := NewGoGetFetcher(s.goBinaryName, fs, repoURI, version)
+	r.NoError(err)
+	ref, err := fetcher.Fetch(repoURI, version)
+	r.NoError(err)
+	ver, err := ref.Read()
+	r.NoError(err)
+
+	r.True(len(ver.Info) > 0)
+
+	r.True(len(ver.Mod) > 0)
+
+	zipBytes, err := ioutil.ReadAll(ver.Zip)
```

Comment:
does `ver.Zip` still need to be closed?

Reply:
yes, it does. for a follow-up PR, do you have any ideas for making this simpler? right now, `fetcher.Fetch()` closes the info and mod files, but leaves the zip file open and forces you to close it for you. that feels a little weird and maybe like a leaky abstraction...

I don't want to block this PR on that if that's ok with everyone here, but I'm wondering if we can do better in a follow-up

Reply:
For sure. I'm sure there will be lots of iterations on the tests moving forward.